src/reporters.ts

```diff
@@ -12,22 +12,41 @@ import { pipe } from "fp-ts/lib/function";
  *   ".a is not a number"
  *   ".c.b is not a non empty string"
  */
+
 const getContextPath = (context: Context): string => {
   if (context.length === 0) {
-    return "] (decoder info n/a)";
+    return " (decoder info n/a)";
   }
-  const keysPath = context.map(({ key }) => key).join(".");
+
+  /*
+    in order to get a more readable ouptut we want to avoid those keys that represent the index of a t.union or a t.intersection, 
+    to do this we want to render the key in the output just in 2 cases: 
+      1. the key can be parsed as a valid integer and the element right before was an ArrayType
+      2. the key is not parsable as a valid integer but it is also a non empty string
+  */
+  const keysPath = context
+    .filter(
+      // eslint-disable-next-line
+      (c: any, i: number) =>
+        (!isNaN(parseInt(c.key, 10)) &&
```

Comment:
you may want to alias `isNaN(parseInt)` to make this a little more readable (ie. `isInteger`) Another way would be to use` IntegerFromString.is()` maybe?

Reply:
Yess you are right, i didn't notice the `IntegerFromString` type mb, at this point i think that the cleanest way to handle this is to use 

```ts 
IntegerFromString.is(parseInt(c.key, 10))
```

Reply:
I dont' think you need the parseInt part, check if `IntegerFromString.is(c.key)` suffices.

Reply:
> check if `IntegerFromString.is(c.key)` suffices

Unfortunately we need to pass a number to  `is()` otherwise it will always return false, it is just a `t.number.is` as you can se here:
```ts 
export const NumberFromString = new t.Type<number, string>(
  "NumberFromString",
  t.number.is,
  (m, c) =>
    pipe(
      t.string.validate(m, c),
      E.chain(s => {
        const n = +s;
        return isNaN(n) ? t.failure(s, c) : t.success(n);
      })
    ),
  String
);
```

Reply:
You can use `IntegerFromString.is(+c.key)` as a shourtcut.

Reply:
Fine, I'm going for the `IntegerFromString.is(+c.key)` way 😃